src/js/containers/covid19/recipient/SpendingByRecipientContainer.jsx

```diff
@@ -0,0 +1,233 @@
+import ResultsTableErrorMessage from 'components/search/table/ResultsTableErrorMessage';
+
+const propTypes = {
+    activeTab: PropTypes.string.isRequired
+};
+
+const columns = [
+    {
+        title: 'recipient',
+        displayName: 'Recipient'
+    },
+    {
+        title: 'obligation',
+        displayName: 'Award Obligations',
+        right: true
+    },
+    {
+        title: 'outlay',
+        displayName: 'Award Outlays',
+        right: true
+    },
+    {
+        title: 'count',
+        displayName: (
+            <>
+                <div>Number</div>
+                <div>of Awards</div>
+            </>
+        ),
+        right: true
+    }
+];
+
+const loanColumns = [
+    {
+        title: 'recipient',
+        displayName: 'Recipient'
+    },
+    {
+        title: 'faceValue',
+        displayName: (
+            <>
+                <div>Face Value</div>
+                <div>of Loans</div>
+            </>
+        )
+    },
+    {
+        title: 'obligation',
+        displayName: (
+            <>
+                <div>Award Obligations</div>
+                <div>(Loan Subsidy Cost)</div>
+            </>
+        ),
+        right: true
+    },
+    {
+        title: 'outlay',
+        displayName: (
+            <>
+                <div>Award Outlays</div>
+                <div>(Loan Subsidy Cost)</div>
+            </>
+        ),
+        right: true
+    },
+    {
+        title: 'count',
+        displayName: (
+            <>
+                <div>Number</div>
+                <div>of Awards</div>
+            </>
+        ),
+        right: true
+    }
+];
+
+export const parseRows = (rows, activeTab) => (
+    rows.map((row) => {
+        const rowData = Object.create(CoreSpendingTableRow);
+        rowData.populateCore(row);
+        let link = rowData.description;
+        if (rowData._id) {
+            link = (
+                <a href={`#/recipient/${rowData._id}`}>
+                    {rowData.description}
+                </a>
+            );
+        }
+        if (activeTab === 'loans') {
+            return [
+                link,
+                rowData.faceValue,
+                rowData.obligation,
+                rowData.outlay,
+                rowData.count
+            ];
+        }
+        return [
+            link,
+            rowData.obligation,
+            rowData.outlay,
+            rowData.count
+        ];
+    })
+);
+
+const SpendingByRecipientContainer = ({ activeTab }) => {
+    const [currentPage, changeCurrentPage] = useState(1);
+    const [pageSize, changePageSize] = useState(10);
+    const [totalItems, setTotalItems] = useState(0);
+    const [results, setResults] = useState([]);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(false);
+    const [sort, setSort] = useState('obligation');
+    const [order, setOrder] = useState('desc');
```

Comment:
we should have a custom pagination hook

```javascript
const usePagination = (initialState) => {
    const [currentPage, changeCurrentPage] = useState(1);
    const [pageSize, changePageSize] = useState(10);
    const [totalItems, setTotalItems] = useState(0);
    const [results, setResults] = useState([]);
    const [loading, setLoading] = useState(true);
    const [error, setError] = useState(false);
    const [sort, setSort] = useState('obligation');
    const [order, setOrder] = useState('desc');
  // ?
  return [];
}
```

☝️  not sure how this would work but it would be nice.

Reply:
Agreed, that would be great since these state vars get repeated all over!

Reply:
how would we do this though... im curious as to how we would update an object like pagination using hooks. Do we do that anywhere?